### ISSUE TITLE
Failure cluster [9a6a5067...] [sig-cli] Kubectl client [k8s.io] Guestbook application should create and stop a working application [Conformance]

### ISSUE THREAD

**Issue by dims** [kubernetes/kubernetes#73111]
##### Error text:
```
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/framework.go:699
Expected error:
    <exec.CodeExitError>: {
        Err: {
            s: "error running &{/google-cloud-sdk/bin/kubectl [kubectl --kubeconfig=/tmp/kops863247222/kubeconfig create -f - --namespace=kubectl-4679] []  0xc0018e1460  error: SchemaError(io.k8s.api.admissionregistration.v1beta1.MutatingWebhookConfiguration): invalid object doesn't have additional properties\n [] <nil> 0xc001941560 exit status 1 <nil> <nil> true [0xc001c0c9e8 0xc001c0ca10 0xc001c0ca20] [0xc001c0c9e8 0xc001c0ca10 0xc001c0ca20] [0xc001c0c9f0 0xc001c0ca08 0xc001c0ca18] [0x9178c0 0x9179c0 0x9179c0] 0xc0018dc780 <nil>}:\nCommand stdout:\n\nstderr:\nerror: SchemaError(io.k8s.api.admissionregistration.v1beta1.MutatingWebhookConfiguration): invalid object doesn't have additional properties\n\nerror:\nexit status 1\n",
        },
        Code: 1,
    }
    error running &{/google-cloud-sdk/bin/kubectl [kubectl --kubeconfig=/tmp/kops863247222/kubeconfig create -f - --namespace=kubectl-4679] []  0xc0018e1460  error: SchemaError(io.k8s.api.admissionregistration.v1beta1.MutatingWebhookConfiguration): invalid object doesn't have additional properties
     [] <nil> 0xc001941560 exit status 1 <nil> <nil> true [0xc001c0c9e8 0xc001c0ca10 0xc001c0ca20] [0xc001c0c9e8 0xc001c0ca10 0xc001c0ca20] [0xc001c0c9f0 0xc001c0ca08 0xc001c0ca18] [0x9178c0 0x9179c0 0x9179c0] 0xc0018dc780 <nil>}:
    Command stdout:
    
    stde
```
#### Recent failures:
[1/19/2019, 8:15:38 PM ci-kubernetes-e2e-kops-gce-canary](https://gubernator.k8s.io/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-kops-gce-canary/7436#sig-cli-kubectl-client-k8sio-guestbook-application-should-create-and-stop-a-working-application-conformance)
[1/19/2019, 7:27:39 PM ci-kubernetes-e2e-kops-gce-canary](https://gubernator.k8s.io/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-kops-gce-canary/7435#sig-cli-kubectl-client-k8sio-guestbook-application-should-create-and-stop-a-working-application-conformance)
[1/19/2019, 6:27:47 PM ci-kubernetes-e2e-kops-gce](https://gubernator.k8s.io/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-kops-gce/17933#sig-cli-kubectl-client-k8sio-guestbook-application-should-create-and-stop-a-working-application-conformance)
[1/19/2019, 5:05:38 PM ci-kubernetes-e2e-kops-gce-canary](https://gubernator.k8s.io/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-kops-gce-canary/7431#sig-cli-kubectl-client-k8sio-guestbook-application-should-create-and-stop-a-working-application-conformance)
[1/19/2019, 4:03:46 PM ci-kubernetes-e2e-kops-gce](https://gubernator.k8s.io/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-kops-gce/17929#sig-cli-kubectl-client-k8sio-guestbook-application-should-create-and-stop-a-working-application-conformance)

/label sig/cli

**Comment by dims** [comment]
possibly related to https://github.com/kubernetes/test-infra/issues/10850

**Comment by Shnatsel** [comment]
/kind test-failure
@kubernetes/sig-testing

**Comment by justinsb** [comment]
I think this is related https://github.com/kubernetes/kubernetes/pull/56434 , caused by kubectl vs server skew.  That's based on the error message, but I haven't yet found the actual code that is the source of the error (i.e. the fmt.Errorf statement).

**Comment by spiffxp** [comment]
/milestone v1.14
We're entering [burndown](https://github.com/kubernetes/sig-release/tree/master/releases/release-1.14#burndown) and this looks relevant to the 1.14 release. Please `/milestone clear` if I am incorrect

**Comment by spiffxp** [comment]
https://storage.googleapis.com/k8s-gubernator/triage/index.html?test=Guestbook%20application%20should%20create%20and%20stop%20a%20working%20application#%7Bid%7D

Couldn't find the existing cluster.

**Comment by soggiest** [comment]
@kubernetes/sig-cli-maintainers We're in 1.14 code freeze and this looks to be relevant for the 1.14 release. Can we get some more eyes on this?

**Comment by alejandrox1** [comment]
Please correct me if I'm wrong, but the test failure mentioned in this issue, `[sig-cli] Kubectl client [k8s.io] Guestbook application should create and stop a working application [Conformance] 6.73s` doesn't appear in the sig-release dashboards so I will take this one off the CI signal project.

**Comment by toyota790** [comment]
I have also got the similar issue when I execute the kubectl create or apply command on version 1.14.0-beta.1 of K8S cluster. 
It showed

>  kubectl apply -f iis.yaml 
error: SchemaError(io.k8s.api.autoscaling.v2beta1.PodsMetricSource): invalid object doesn't have additional properties

**Comment by sandeep-krishnan** [comment]
> I also got the similar issue when I execute the kubectl create or apply command on version 1.14.0-beta.1 of K8S cluster.

+1

**Comment by Ademord** [comment]
As per [source](https://stackoverflow.com/questions/55417410/kubernetes-create-deployment-unexpected-schemaerror):

> It was caused by the version mismatch of kubectl that was mentioned a couple of times already in this thread. Docker installs version 1.10 of kubectl. 

Which can be verified at: ❌
```
> kubectl version
Client Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.11", GitCommit:"637c7e288581ee40ab4ca210618a89a555b6e7e9", GitTreeState:"clean", BuildDate:"2018-11-26T14:38:32Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.0", GitCommit:"641856db18352033a0d96dbc99153fa3b27298e5", GitTreeState:"clean", BuildDate:"2019-03-25T15:45:25Z", GoVersion:"go1.12.1", Compiler:"gc", Platform:"linux/amd64"}
```
On Mac, you can run:
```
    rm /usr/local/bin/kubectl
    brew link --overwrite kubernetes-cli
```
Then I applied my ```kubectl apply``` , which worked. 🎉

And now ✅
```
> kubectl version
Client Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.0", GitCommit:"641856db18352033a0d96dbc99153fa3b27298e5", GitTreeState:"clean", BuildDate:"2019-03-26T00:04:52Z", GoVersion:"go1.12.1", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.0", GitCommit:"641856db18352033a0d96dbc99153fa3b27298e5", GitTreeState:"clean", BuildDate:"2019-03-25T15:45:25Z", GoVersion:"go1.12.1", Compiler:"gc", Platform:"linux/amd64"}
```

Hope this helps 😄

**Comment by MichaelChirico** [comment]
> invalid object doesn't have additional properties

Ace-level useless error message 😂 

Thanks Ademord, Mac solution worked for me 👍

**Comment by awadodeh** [comment]
> As per [source](https://stackoverflow.com/questions/55417410/kubernetes-create-deployment-unexpected-schemaerror):
> 
> > It was caused by the version mismatch of kubectl that was mentioned a couple of times already in this thread. Docker installs version 1.10 of kubectl.
> 
> Which can be verified at: ❌
> 
> ```
> > kubectl version
> Client Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.11", GitCommit:"637c7e288581ee40ab4ca210618a89a555b6e7e9", GitTreeState:"clean", BuildDate:"2018-11-26T14:38:32Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"darwin/amd64"}
> Server Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.0", GitCommit:"641856db18352033a0d96dbc99153fa3b27298e5", GitTreeState:"clean", BuildDate:"2019-03-25T15:45:25Z", GoVersion:"go1.12.1", Compiler:"gc", Platform:"linux/amd64"}
> ```
> On Mac, you can run:
> 
> ```
>     rm /usr/local/bin/kubectl
>     brew link --overwrite kubernetes-cli
> ```
> Then I applied my `kubectl apply` , which worked. 🎉
> 
> And now ✅
> 
> ```
> > kubectl version
> Client Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.0", GitCommit:"641856db18352033a0d96dbc99153fa3b27298e5", GitTreeState:"clean", BuildDate:"2019-03-26T00:04:52Z", GoVersion:"go1.12.1", Compiler:"gc", Platform:"darwin/amd64"}
> Server Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.0", GitCommit:"641856db18352033a0d96dbc99153fa3b27298e5", GitTreeState:"clean", BuildDate:"2019-03-25T15:45:25Z", GoVersion:"go1.12.1", Compiler:"gc", Platform:"linux/amd64"}
> ```
> Hope this helps 😄

This worked for me as well on mac! Much appreciated!

**Comment by vinothrao** [comment]
For windows 

This has happened to me because of wrong version of kubectl installed by docker in the path
"C:\Program Files\Docker\Docker\resources\bin" .

I resolved this by following below steps

Install kubectl manually by :

Install-Script -Name install-kubectl -Scope CurrentUser -Force
install-kubectl.ps1 [-DownloadLocation ]

(https://kubernetes.io/docs/tasks/tools/install-kubectl/#install-kubectl-on-windows)

CD to the [-DownloadLocation ] folder and run kubectl from here . If the location is not given it will install it in temp by default. I have installed it in temp
PS C:\Users\sysadmin\AppData\Local\Temp> : .\kubectl.exe version

If the version mismatch run the command again and check.
Try kubectl apply from this location.

PS C:\Users\sysadmin\AppData\Local\Temp> .\kubectl.exe apply -f C:\Users\sysadmin\source\repos\RecruitManage\webapp-pod.yaml

Note : careful about that kubectl use .\kubectl.exe which is inside the folder not the plain kubectl because it will fall back to Docker's kubectl as it is set in environment variable.
